executor/merge_join.go

```diff
@@ -109,56 +117,81 @@ func (t *mergeJoinInnerTable) init(ctx context.Context, chk4Reader *chunk.Chunk)
 	return err
 }
 
-func (t *mergeJoinInnerTable) rowsWithSameKey() ([]chunk.Row, error) {
-	lastResultIdx := len(t.resultQueue) - 1
-	t.resourceQueue = append(t.resourceQueue, t.resultQueue[0:lastResultIdx]...)
-	t.resultQueue = t.resultQueue[lastResultIdx:]
+func (t *mergeJoinInnerTable) selectRow(row chunk.Row) {
+	t.curSel = append(t.curSel, row.Idx())
+}
+
+func (t *mergeJoinInnerTable) selectedRows() *chunk.RowContainer {
+	if t.curSel != nil {
+		t.curChk.SetSel(t.curSel)
+		t.sameKeyRows.SetTemporary(t.curChk)
```

Comment:
How about implement a multiple iterator like [`io.MultiReader`](https://golang.org/pkg/io/#MultiReader) to avoiding this confusing `SetTemporary` method?

Reply:
Sounds good, maybe next PR....

Reply:
This method lets me feel confused too... How about adding a TODO comment here?